lib/util/http/index.ts

```diff
@@ -0,0 +1,118 @@
+import is from '@sindresorhus/is/dist';
+import URL from 'url';
+import got from '../got';
+
+interface OutgoingHttpHeaders {
+  [header: string]: number | string | string[] | undefined;
+}
+
+export interface HttpOptions {
+  auth?: string;
+  baseUrl?: string;
+  headers?: OutgoingHttpHeaders;
+  throwHttpErrors?: boolean;
```

Comment:
This is only used by docker datasource. We could adapt it to catch the error instead.